### PR TITLE
CAI Integration - Installer

### PR DESCRIPTION
**Pull request by joecheuk**

### CODE REVIEW

File: install/gcp/installer/util/gcloud.py [hunk 1]
```diff
@@ -279,8 +337,7 @@ def _grant_svc_acct_roles(target_id,
     if grant_roles_cmds:
         print(constants.MESSAGE_CREATE_ROLE_SCRIPT)
 
-        with open('grant_forseti_roles.sh', 'wt') as roles_script:
-            roles_script.write('#!/bin/bash\n\n')
```
Comment (blueandgold) [inline install/gcp/installer/util/gcloud.py]: Why is this being removed?

File: install/gcp/installer/util/gcloud.py [hunk 2]
```diff
@@ -249,10 +251,66 @@ def grant_server_svc_acct_roles(enable_write,
...
+        user_can_grant_roles)
+
+    has_role_script_rest = _grant_svc_acct_roles(
         target_id, project_id, gcp_service_account,
         user_can_grant_roles, roles)
 
+    return has_role_script_bucket or has_role_script_rest
+
+
+def _grant_bucket_obj_roles(gcp_service_account,
```
Comment (blueandgold) [inline install/gcp/installer/util/gcloud.py]: The `bucket_obj` naming here can be confusing, with the actual storage object.  Can we just rename to `_grant_bucket_roles()`?

File: install/gcp/installer/util/gcloud.py [hunk 3]
```diff
@@ -249,10 +251,66 @@ def grant_server_svc_acct_roles(enable_write,
...
+                            bucket_name,
+                            roles_to_grant,
+                            user_can_grant_roles):
+    """Grant GCS object roles to GCP service account.
+    Note: This is only supported through gsutil library and not in
```
Comment (blueandgold) [inline install/gcp/installer/util/gcloud.py]: newline here

File: install/gcp/installer/util/gcloud.py [hunk 4]
```diff
@@ -249,10 +251,66 @@ def grant_server_svc_acct_roles(enable_write,
...
+        roles_to_grant (list): List of roles to grant.
+    Returns:
+        bool: Whether or not a role script has been generated.
+    """
+    failed_commands = []
+
+    for role in roles_to_grant:
+        member = 'serviceAccount:{}:{}'.format(gcp_service_account, role)
+        bucket = 'gs://{}'.format(bucket_name)
+        bucket_obj_role_cmd = ['gsutil', 'iam', 'ch', member, bucket]
```
Comment (blueandgold) [inline install/gcp/installer/util/gcloud.py]: ditto: on removing `obj` here

File: install/gcp/installer/util/gcloud.py (hunk 2 above)
Comment (joecheuk) [reply install/gcp/installer/util/gcloud.py]: Updated.

File: install/gcp/installer/util/gcloud.py (hunk 3 above)
Comment (joecheuk) [reply install/gcp/installer/util/gcloud.py]: Updated.

File: install/gcp/installer/util/gcloud.py (hunk 4 above)
Comment (joecheuk) [reply install/gcp/installer/util/gcloud.py]: Updated.

File: install/gcp/installer/util/gcloud.py (hunk 1 above)
Comment (joecheuk) [reply install/gcp/installer/util/gcloud.py]: updated.

**Comment by ahoying** [comment]
I think we should also set the retention period for the CAI bucket to delete all files older than 14 days, plenty of time to troubleshoot, but then we don't have a constantly filling bucket costing the customers money (and note this in the documentation as well).

https://cloud.google.com/storage/docs/gsutil/commands/lifecycle

cai-bucket-lifecycle.json:
{
  "rule":
  [
    {
      "action": {"type": "Delete"},
      "condition": {"age": 14}
    }
  ]
}

gsutil lifecycle set cai-bucket-lifecycle.json gs://{CAI_EXPORT_BUCKET}

File: configs/server/forseti_conf_server.yaml.in [hunk 5]
```diff
@@ -62,8 +62,8 @@ inventory:
           period: 1.1
 
     cai:
-        enabled: False
-        gcs_path: ""
+        enabled: True
```
Comment (ahoying) [inline configs/server/forseti_conf_server.yaml.in]: Should we make this a variable as well? So it can be set false if the root is not an organization?

**Comment by ahoying** [comment]
You can probably set the lifecycle in the deployment manager template as well.

**Comment by joecheuk** [comment]
Added bucket life cycle in deployment manager, will test it out before merging.

**Pull request merged**